src/hypothesis/strategies.py

```diff
@@ -1139,16 +1131,14 @@ def decimals(min_value=None, max_value=None,
             # This could be positive infinity, quiet NaN, or signalling NaN
             raise InvalidArgument(u'Invalid min_value=%r' % min_value)
     if max_value is not None:
-        max_value = Decimal(max_value)
+        max_value = try_convert(Decimal, max_value, 'max_value')
```

Comment:
[This test always passed](https://travis-ci.org/HypothesisWorks/hypothesis-python/jobs/267877445#L2403)